hook/hook.go

```diff
@@ -24,7 +24,18 @@ import (
 )
 
 // Points define user-injected hook points.
+//
+// Directly using this interface as a [BlockBuilder] is indicative of this node
+// locally building a block. Calling [Points.BlockBuilder] with an existing
+// block is indicative of this node reconstructing a block built elsewhere
+// during verification.
 type Points interface {
+	BlockBuilder
+	// BlockBuilder returns a [BlockBuilder] that will attempt to reconstruct
+	// the provided block. If the provided block is valid for inclusion, then
+	// the returned builder MUST be able to reconstruct an identical block.
+	BlockBuilder(block *types.Block) BlockBuilder
```

Comment:
```suggestion
	BlockRebuilderFrom(block *types.Block) BlockBuilder
```

Or just `BlockRebuilder()` although I like how the `From` is very suggestive of intended usage. You'll also need to update the comment on `Points`.